The gross check is missing  for the q observations from the rawinsonde

gmao-wgu opened the pull request:
## Description

*There is no gross check for the q observations from the rawinsonde*

gmao-wgu committed:
added the gross check for q obs from rawinsonde

rtodling reviewed (commented):
I would expect this to come with an update to the benchmark numbers in ufo_tests.yaml ... I'd consider suspicious if the sondes numbers does change with  this ... !?

gmao-wgu commented:
It's odd that we were able to get matching numbers between GSI and JEDI without the gross check included. @asewnath what is your take on this?

gmao-wgu commented:
I am still looking into it.

Dooruk commented:
I ran a Tier1 test with this branch and ufo is indeed failing:
https://github.com/GEOS-ESM/swell/actions/runs/10947010213

Looks like it's a minor diff though, @gmao-wgu if you want to change the value to 116684:
`FAILED passed == passedBenchmark evaluated as [116684] == [116687]`

https://github.com/GEOS-ESM/swell/blob/dae29b4b5f67e0dd344af8a2ad70de67b56ef8cc/src/swell/configuration/jedi/interfaces/geos_atmosphere/observations/ufo_tests.yaml#L135-L137

asewnath commented:
> I ran a Tier1 test with this branch and ufo is indeed failing: https://github.com/GEOS-ESM/swell/actions/runs/10947010213
> 
> Looks like it's a minor diff though, @gmao-wgu if you want to change the value to 116684: `FAILED passed == passedBenchmark evaluated as [116684] == [116687]`
> 
> https://github.com/GEOS-ESM/swell/blob/dae29b4b5f67e0dd344af8a2ad70de67b56ef8cc/src/swell/configuration/jedi/interfaces/geos_atmosphere/observations/ufo_tests.yaml#L135-L137

Thanks for checking this @Dooruk!

Dooruk committed:
Merge branch 'develop' into feature/grosscheck4q

Dooruk committed:
update test value

gmao-wgu commented:
This is still incomplete. I have not figured out how to set the lower and upper bounds for the q observation errors.

Dooruk commented:
> This is still incomplete. I have not figured out how to set the lower and upper bounds for the q observation errors.

Sounds good. I thought the UFO value was the final change required.

ashiklom committed:
Remove hardcoded /tmp from code_tests, and fix GEOS_mksi test (#436)

* Remove hardcoded /tmp from code_tests

See #432

* Fix GEOS_mksi test

Since https://github.com/GEOS-ESM/GEOS_mksi/pull/2 was merged, GEOS_mksi
main and develop are now in sync. This was causing the GEOS_mksi test
to fail.

This changes the test to point to a known (very old!) version of
GEOS_mksi that will cause an error, so the test actually passes.

Dooruk committed:
Cycling FGAT for `geos_marine` and towards erasing `geos_ocean` (#419)

* executable and a brand new task

* new task and dynamic keys as a new option

* make suite to run accesible

* soft run option for generate B climatology

* fgat cycle suite and template yamls

* eva change

* codestyle fixes

* allow setting env variable with code tests

* add fgat executable and reorganize linking according to models

* code test fixes

* create marine_models key as a SOCA related tasks control switch

* use marine_models in templates to control config behavoir

* fix default typehinting

* update config to use generic saber

* adapt to using cwd, include **kwargs in shell utils

* update link_geos for cice6 history handling

* fgat with geos_marine

* fix typehinting

* include analysis time as a template variable

* eva related changes for increment outputs

* preaching docs, gotta add docs

asewnath committed:
Using Pinned Hashes for JEDI Repositories  (#433)

* feat: add pinned hashes options for jedi bundle

Dooruk committed:
Adding Instructions on how to use `uv` and `venv` for Swell installation (#446)

* alternative way of swell installation

* minor fixes

* clarification

rtodling reviewed (approved):
I'm fine w/ this.

Dooruk commented:
@gmao-wgu is this ready to go in? needs to be merged with develop

gmao-wgu commented:
@Dooruk It is still a work in progress because the upper and lower bounds of the observation error are missing in the gross check.

Dooruk commented:
> @Dooruk It is still a work in progress because the upper and lower bounds of the observation error are missing in the gross check.

Sounds good, thanks

rtodling committed:
Add hydrometeors (#422)

* add qr/qs

* w/o this code defaults to GFDL settings

* a bit more more hydrometeors

* no reason for these to be exec files

* null getvalues for marine

---------

Co-authored-by: dooruk <dorukhan.ardag@nasa.gov>

mranst committed:
Initial commit - fixed incorrect definition of marine_models in 3dfgat_cycle (#448)

Co-authored-by: Doruk Ardağ <38666458+Dooruk@users.noreply.github.com>

metdyn committed:
LGETKF: add function to tune horizontal localization for each observer (#437)

* Add interfaces/geos_atmosphere/observations/localization

* - run_jedi_local_ensemble_da_executable.py can use yaml set in observations/localization
- add observations/localization

* removed *log  from clean_patterns: in suite_tests/localensembleda-tier1.yaml

* Separate conv obs from Sat obs to test LGETKF

* .

* Add in more observations to localensembleda-tier1.yaml

* python code test pass

* fix ntasks per node for SLES15

* fix mock test

---------

Co-authored-by: Doruk Ardağ <38666458+Dooruk@users.noreply.github.com>

rtodling committed:
Oct 2024 Variable Change Sprint (#457)

* this now has Nov19 - past sprint version working for sondes

* these now take care of most of the conv obs

* not quite sure why we have these files here!

* this works now

* work now - the ObsDiags commented out here always caused an odd warning - now it causes a crash

* this now runs, but the JoJc become negative as the iterations go one ... I wonder if this was the case in the original code, but had gone unnoticed - I am checking

* updated geovals location w/ file that incorporate new variable names

* fix for aircraft; update obs-exp location (will need another update)

* change ocean variable names

* change obs reference to x048v3

* presently - circa 3 Dec using 19 Nov JEDI - this no longer works in post filters

* eva changes for ice

* seaice related variable name changes

* seaice variables renaming

* no need to overwrite defaul obs abd geovals paths

* take out overwrite from the ufo testing

---------

Co-authored-by: dooruk <dorukhan.ardag@nasa.gov>

rtodling committed:
add missing surface type to linear obs op of MW (#463)

rtodling committed:
tropopause pressure consistent w/ GSI (#464)

Dooruk committed:
Introducing GMAO ODAS obs sources (#444)

* initial changes

* fix calc_scales and vertical corr. length configuration

* add first set of ODAS obs

* more obs

* fix get_obs for multiple providers, missing obs, and add marine yamls

* add cryosat and sentinel3b

* add MERRA2 SST

* fix blacklist, simplify generate B task

* fix for using obs_providers as a list

* fix code issues

* take two

Dooruk committed:
Keep the order of YAML configurations within run tasks (#485)

* keep the order recursively

* codestyle fix

metdyn committed:
A draft for obs thinning (#482)

* Add interfaces/geos_atmosphere/observations/localization

* - run_jedi_local_ensemble_da_executable.py can use yaml set in observations/localization
- add observations/localization

* removed *log  from clean_patterns: in suite_tests/localensembleda-tier1.yaml

* Separate conv obs from Sat obs to test LGETKF

* .

* Add in more observations to localensembleda-tier1.yaml

* python code test pass

* ...

* use c12 and 2 member

* ...

* First trial for framework to work

* v1. run_jedi_obsfilters_executable.py
Error:
  run_jedi_obsfilters_executable.py", line 33, in execute
      window_type = self.config.window_type()

* .

* .

* Add file_orig.nc

* Trial: replace run_jedi_obsfilters_executable.py by run_jedi_hofx_executable.py
Next: observer is missing jedi_hofx_config.yaml
      merge develop and test why

* test

* Save a copy for dict.clear  dict.update
Next: run with np=1 for obs thinning

* updated observations for obs thinning

* recover a few files from develop

* Add correct yaml content for obs thinniong section

* ...

* still debug

* No eror if use MPI:
mpirun -np 1 /discover/nobackup/projects/gmao/aist-nr/yyu11/br_jedi_scu17/run_swell/qc3/jedi_bundle/build/bin/test_ObsFilters.x \
/discover/nobackup/projects/gmao/aist-nr/yyu11/br_jedi_scu17/run_swell/qc3/run/20211212T000000Z/geos_atmosphere/jedi_obsfilters_config.yaml \
./jedi_obsfilters_log.log

Get HDF5 error using pypthon subprocess:
                command = ["mpirun", "-np", "1", jedi_executable_path, jedi_config_file, output_log_file]
                result = subprocess.run(command, check=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)

* update

* Surprisingly, I have to move in the localensembleda-tier1.yaml
 - amsua_metop-b
 to the end of the list

* improved code style

* fix unused variables

* update

mranst committed:
Platform for cascade lake nodes running sles15 (#488)

* Initial commit

* changed modules to use scu17

rtodling committed:
Revise obs space localization for LETKF (#477)

* revise obs space localization for LETKF

* use all ob system

* this was missing

* works but ...

* Help with PR477 (#479)

* ...

* Add to localensembleda-tier1.yaml
    local_ensemble_save_posterior_ensemble: false
    local_ensemble_save_posterior_ensemble_increments: false
    local_ensemble_save_posterior_mean: false
    local_ensemble_save_posterior_mean_increment: true
output jedi_localensembleda_config.yaml shows right.

* In run_jedi_local_ensemble_da_executable.py, hard code if bypass_writing_hofxs:  del [obs space][obsdataout]

* update default for save_posterior_ensemble

* add back lienar_observer: true

---------

Co-authored-by: Yonggang Yu <yonggang.g.yu@gmail.com>
Co-authored-by: Yonggang Yu <yonggang.yu@nasa.gov>

mranst committed:
Ability to run tier 2 tests locally (#481)

* Tier 2 test support for user yaml, added documentation

* Support for tier 2 tests from command line

* Added documentation for tier2 tests and platform choice

* removed leftover print

* Added capability to use tier2 overrides from suite_tests

* string format fix

* create_experiment now uses tier2 overrides if file is present

* Simplified tier test selection, removed print calls from debug

* fixed call to tier2 enum

* fixes to suite tests

* removed unused function

rtodling committed:
These cause Jo to go negative (#460)

Dooruk committed:
Adding EMC L3 SST, getting rid off obsolete observers (#492)

* add EMC SST retrievals

* satellite names

* erase obsolete observers

* change adt config filters, add marine component switches

* sst config update, switch to absolute threshold

* sss changes

* make changes in tasks so marine_models switch can function

* abandon old marine observers

* version bump

* erase unused obs

* fix merra2 filters

* set obs provider s2s_v1 for marine and ocean, remove obsolete observer references

* fix hofx test obs provider

DavidSRussell committed:
Allow for assimilation of SWOT Nadir altimetry (#494)

* Allow for assimilation of SWOT nadir altimetry in geos_marine

* Allow for assimilation of SWOT nadir altimetry in geos_ocean

* Remove old obs error assignment from adt_swot_nadir.yaml

* Remove old blacklists from adt_swot_nadir.yaml

* Update sea_ice_area_fraction handling in adt_swot_nadir.yaml

* Update minimum SST value in adt_swot_nadir.yaml

---------

Co-authored-by: Doruk Ardağ <38666458+Dooruk@users.noreply.github.com>

rtodling committed:
Peg mksi to tag (#506)

ashiklom committed:
Fix broken links in docs (#500)

* Fix broken links in docs

* Add header to "history outputs" docs page

So that it renders more cleanly.

mranst committed:
Upgrade modules to spack-stack 1.9, and update JEDI for February 6 (#504)

* Initial commit

* updated cylc instructions for spack-stack 1.8

* Change sles15 to skylab9

* Update docs for installation using spack-stack 1.9, python 3.11

* Update existing JEDI build to Feb 06

* Fixed a potential issue where ~ is not resolved for mksi_path

* Update pinned_versions to be consistent with  02/06 version of jedi_bundle

* include cascade module updates, take out ecflow (#508)

* Peg mksi to tag (#506)

* add cascade modules, take out ecflow

* revert solo module change

---------

Co-authored-by: rtodling <ricardo.todling@nasa.gov>

* update cascade JEDI builds

---------

Co-authored-by: Doruk Ardağ <38666458+Dooruk@users.noreply.github.com>
Co-authored-by: rtodling <ricardo.todling@nasa.gov>
Co-authored-by: dooruk <dorukhan.ardag@nasa.gov>

mranst committed:
Swell logger extends logging.Logger (#450)

* now use get_logger function

* Added comments for changes to logger class

* Refactor logger for consistency with logging library

* Logging refactor compatible with code tests

* Suppress warning from NUMEXPR

* Add comment clarifying magic number for logger

* pycodestyle fix

mranst committed:
Fixed bug where models in override.yaml would overwrite tier1 defaults (#511)

Dooruk committed:
Reviving 3dvar_cycle and 3dfgat_cycle suite defaults (#512)

* update default geos folders

* version bump

* get tile.bin produced in the folder

* fix tier1 defaults

* fix for when marine_models is None

* eva for geos_ocean

* soca2cice is cice6 dependent

* soca2cice related changes

* take out irrelevant parts

* make sea-ice related geos_marine configurations cice6 conditional

* empty line end of the line

* eva fix for atmosphere

* soca2cice requires marine_models key

* marine_models flag for ostia

rtodling committed:
AVHRR - as all IR - needs all surf types in linear model (#468)

Co-authored-by: Doruk Ardağ <38666458+Dooruk@users.noreply.github.com>

rtodling committed:
Add perhost opt for mpi execs (#517)

* Add perhost opt for mpi execs

* fix attempt to get perhost from config

* show perhost in log

* Update src/swell/utilities/run_jedi_executables.py

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>

* Update src/swell/tasks/run_jedi_variational_executable.py

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>

* Update src/swell/utilities/run_jedi_executables.py

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>

* compilation fix

---------

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>

Dooruk committed:
Multiple fixes, save_restart, abandon sles12, another example in the documentation (#516)

* dataatm fix

* update EVA plots for sea-ice increments

* background filters for insitu obs

* fix for missing output obs and EVA to be executed on compute nodes

* fix for history output shenanigans

* messy task

* save restart and increments to R2D2 folder

* don't use _rst for forecast begin time

* take out sles12 instruction

* add new example for 3dfgat_cycle

* special handling for mom6_increment

* eva fixes for sea ice domain

* bye bye sles12

* code_tests related fixes

* fix unused variables

* use spack v1.9 for py_lmod

* fix uv instructions

* save diag outputs while cycling

* fix sea-ice increments

* update docu file

* None perhost fix

Dooruk commented:
@gmao-wgu , is this ready to be merged?

gmao-wgu commented:
@Dooruk Since code changes are involved, it will not work until the code is merged. I have not submitted my PR yet because I am waiting to see if additional changes are needed to handle QC for observations from surface marine and surface land stations. Once this is done, I will submit my PR to include  all these changes together.

ashiklom committed:
Use `null` instead of `None` in YAML (#527)

PyYAML parses `value: None` as `{"value": "None"}` (i.e., the string
`"None"`). In YAML, the correct way to represent a `None` object is one
of the following:

```
value:

value: null

value: ~
```

I'm using `null` here because, in my opinion, it's the clearest one.

Closes #524.

metdyn committed:
Read in execution_time_limit from slurmfile.yaml (#519)

* Read in execution_time_limit from slurmfile.yaml

* update

metdyn committed:
LGETKF with x0050 bkg (#515)

* new configuration with NC4 bias file for aircraft bias coefficients - modified aircraft.yaml
to use the new file, and modified gsi_bc_to_ioda.py to generate the NC4 coeff. file
Working on get_observations.py to get the NC4 aircraft bias coefficient file instead of csv

* change indentation of 'obs bias' section
(it's not a part of obs space)

* some modifications - not yet running

* Try 'satbias' type in get_observations.py (?)
additions for gsi_bc_to_ioda.py - use an aircraft 'sensor'

* remove old 'csv' processing that was left in the missing_obs_test file
change aircraftAscentRate to instantaneousAltitudeRate to conform with
recently changed nomenclature

* Added missing line for appending info on output to yaml
Fixed typo: add 'predictors' not 'default predictors' to output

* Needed to update the predictor coefficient name to match the new
naming convention.  Use 'instantaneousAltitudeRate' and
'instantaneousAltitudeRate_order_2'

* Remove reference to a non-existent file that was pasted into the bias section

* Corrections for python coding norms (indentation, long lines, too
many blank lines)

* Adjust indentation

* Add indentation on two lines for coding standard.  (Hope this works)

* try increase indent again

* tweak indentation again

* ensemble folder

* new aircraft bias format and extension

* fixes to provide _cov file separately for aircraft

* fix test

* reset location of alternative - bkg - files

* split aircraft T/W and adjust yamls  - including commenting filter not working at this time

* needs these too - sorry

* add temp and wind for aircraft but intentinally left aircraft alone

* eva seems to work now

* ...

* Note: I need to delete
 src/swell/configuration/jedi/oops/ensemblehofx3D.yaml
 because fvj hx .cc are in hofx4D

* output files for aircraft varBC

* minor fix

* updates for handling 20231010 00z case

* fix tier1 overrides

* aircraft is two separate entries now

* codestyle

* arrange defaults and overrides for tier1 tests

* make ufo_testing work for x0050

* deactivate ocean for hofx temporarily

* c91 as tier1 test, adapt to geos experiment folder style

* hofx has to use x0050 data

* add geovals for ufo_testing

* update hx and lens .yaml

* recover run_jedi_hofx_executable.py to discover

* ...

* changed  filenames: ['ebkg/mem%mem%/geos.mem%mem%.%yyyy%mm%dd_%hh%MM%ssz.nc4']  for ebkg

* .

* update

* run pycodestyle_run.py

* .

* midifications to enable perhost and execution_time_limit in slurmfile.yaml

* .

* .

* update

* debug run_and_track_subprocess command failed

* .

* fix --perhost N problem in python env

* update mem16  obs22

* Related to
slurm_directives_tasks:
  RunJediLocalEnsembleDaExecutable:
    execution_time_limit: 'PT2H30M'

* recover hofx-tier1.yaml

* corrections

* change after code_tests

* use run_executable in run_jedi_obsfilters_executable.py

* remove subprocess in run_jedi_obsfilters_executable.py

* clean up

* add another posibility for parsing perhost from slurmfile

* separate ens_hofx and eva files from this PR

* fix an error in localensembleda/flow.cylc  `execuation_time_limit`

* .

* .

* .

* .

* scontrol show node borgj192  | grep -i free
   RealMemory=500000 AllocMem=500000 FreeMem=361045 Sockets=2 Boards=1

* modify from PR suggestions and correct code_tests errors

* .

* .

* Add background_time_offset: 'PT3H'  into localensembleda-tier1.yaml

* - recover SaveObsDiags script
- make random seed random in filter_thinning

* revise for code style

---------

Co-authored-by: gmao-msienkie <gmao-msienkie@users.noreply.github.com>
Co-authored-by: dooruk <dorukhan.ardag@nasa.gov>
Co-authored-by: Doruk Ardağ <38666458+Dooruk@users.noreply.github.com>
Co-authored-by: Ricardo Todling <ricardo.todling@nasa.gov>

DavidSRussell committed:
Allow assimilation of Jason-3N (#523)

* Allow assimilation of Jason-3N (Jason-3, new orbit)

* add obs in geos marine

* fix record frequency format

* fix AGCM.rc format

---------

Co-authored-by: dooruk <dorukhan.ardag@nasa.gov>
Co-authored-by: Doruk Ardağ <38666458+Dooruk@users.noreply.github.com>

mranst committed:
A prototype for revised suite and task questions, redux (#514)

* Update question methodology

* Add 3dvar as an example of tier1 defaults, pycodestyle conformance

* Add capability to simply specify model-dependent defaults

* Add enum-based dtype specification for questions

* Consolidate all question defaults into one list

* Consolidate tier1 defaults into suite question lists

* Rename suite defaults list name to base suite name

* Different suite configs can now be called directly

* pycodestyle fix

* Seperate storage of suite defaults into suite files

* Add separate lists for each suite

* Remove tier1 yaml files, pycodestyle fixes

* Use enum classification for question widget types

* Simplify CLI interface

* pycodestyle fix

* Code test fix

* Add new code test for question dictionaries

* Update docs and fixed model issue

* Add user's guide for creating new suites

* Fix support for swell task

* Fix task config

* Fix config.py

* Update docs/adding_a_suite.md

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>

* Update adding_a_suite.md

* simplify language in prepare_config_and_suite, code_test fix

* Update docs/adding_a_suite.md

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>

* Update docs/adding_a_suite.md

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>

* Change override_with_defaults

* Fix missing float option for widgettype

* Update docs/adding_a_suite.md

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>

* Update isinstance checks to use abstract Mapping type

* Revise type hinting

* Remove base from configs

* Update task_questions.py for upcoming PR #516

* New method for dealing with suite configs using enum

* test fixes

* Add comment explaining lambda usage for dataclass

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>

* Update docs/adding_a_suite.md

Co-authored-by: Doruk Ardağ <38666458+Dooruk@users.noreply.github.com>

* code test fix

* Annotation fix

* Fix missing module

* Fix asking questions multiple times

* Add module specification for cli questions, and fix bug with 3dvar cli

* Update Swell version

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>

* Update src/swell/deployment/prepare_config_and_suite/prepare_config_and_suite.py

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>

* Revision for type hinting lists in SwellQuestions

* Simplify question syntax

* classmethod update and test fix

* cls syntax fix

* Suite questions are now visible for tasks

* Fix for tier tests

---------

Co-authored-by: Alexey Shiklomanov <alexey.shiklomanov@gmail.com>
Co-authored-by: Doruk Ardağ <38666458+Dooruk@users.noreply.github.com>

rtodling reviewed (dismissed):
This is good by me.

gmao-wgu committed:
adding filters to discard surface land and marine obs with very large obs errors (#533)

Co-authored-by: Doruk Ardağ <38666458+Dooruk@users.noreply.github.com>

rtodling commented:
@gmao-wgu is this ready for review?

gmao-wgu commented:
@rtodling I will work on the PR for the JEDI code changes, it is not ready yet.

gmao-wgu committed:
adding gross check for q

gmao-wgu committed:
adding gross check for q

gmao-wgu commented:
messed up rebase, and new PR is https://github.com/GEOS-ESM/swell/pull/535
